### PR TITLE
Suppression list search

### PR DESCRIPTION
**Pull request by asgoel**
@ewandennis I haven't added the tests yet, but wanted to make sure I interpreted this particular query properly, since its a bit different than the others. I had to go ahead and modify the behavior of Endpoint.Response since the suppression list search query returns more than just a `results` field. Let me know if I've done this properly and I'll go ahead and add the tests. Thanks!

### CODE REVIEW

File: lib/suppression_list/list_entry.ex [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+  ## Fields
+   - recipient: Email address that was suppressed
+   - type: Type of suppression record (transactional or non_transactional)
+   - source: Source responsible for inserting the list entry. Can be one of
+      "Spam Complaint", "List Unsubscribe", "Bounce Rule", "Unsubscribe Link",
+      "Manually Added", or "Compliance"
+   - description: explanation of suppression
+  """
+
+  defstruct recipient: :required,
```
Comment (ewandennis) [inline lib/suppression_list/list_entry.ex]: Although `transactional` and `non_transactional` are deprecated, they can still turn up in existing records so if we plan to use this module to represent received suppression list entries, I suggest we include them too.

File: lib/suppression_list/search_result.ex [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+  SparkPost representation of a suppression list search result.
+
+  ## Fields
+   - results: List of %SparkPost.SuppressionList.ListEntry{} objects.
+   - links: Links to the first query in the chain and next query to be made for cursor
+      based pagination.
+   - total_count: Total number of results across all pages based on the query params.
+  """
+
+  defstruct results: :required, links: :required, total_count: :required
```
Comment (ewandennis) [inline lib/suppression_list/search_result.ex]: Tiny formatting nitpick: I tend to give each struct field its own line.

**Comment by asgoel** [comment]
@ewandennis made requested changes - mind taking a look?

**Comment by ewandennis** [comment]
lgtm. I was a little concerned about explicitly converting each suppression list entry into a `%SparkPost.SuppressionList.ListEntry` but I got over it after a quick timing comparison :)

This is ready to merge from my perspective. Thanks again @asgoel!

**Comment by asgoel** [comment]
@ewandennis can we get this one merged in? have another waiting on top of this :)

**Comment by ewandennis** [comment]
@asgoel More you say? I'll wait to cut the next release then ;)

**Pull request merged**